### PR TITLE
tidy and rename livecdreboot.pm

### PR DESCRIPTION
**Pull request by MalloZup**
renamed livecdreboot to install_and_reboot
moved grub2 stuff to grub_test.pm
add fix for timeout escape/issue for grub

progress task #9716

### CODE REVIEW

File: tests/installation/grub_test.pm [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# without any warranty.
+
+use strict;
+use base "basetest";
+use testapi;
+
+sub run() {
+    my $self = shift;
+    up_and_down();
+    assert_screen "grub2", 30;
```
Comment (okurz) [inline tests/installation/grub_test.pm]: use default timeout which is also 30

File: tests/installation/grub_test.pm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    send_key 'esc';    # prevent grub2 timeout
```
Comment (okurz) [inline tests/installation/grub_test.pm]: Isn't the up_and_down already preventing the timeout?

File: main.pm [hunk 3]
```diff
@@ -424,6 +424,7 @@ sub load_reboot_tests() {
         loadtest "installation/boot_encrypt.pm";
     }
     if (installyaststep_is_applicable) {
+        loadtest "installation/grub_test.pm";
```
Comment (okurz) [inline main.pm]: the whole file "grub_test.pm" should not be part of this PR which says "rename livecdreboot"

**Comment by okurz** [comment]
* observe the git commit guidelines
* also delete livecdreboot and not just add new file
* observe inline comments

**Comment by MalloZup** [comment]
ok everything is done. i followed the commandments of git-bible. god forgive my sins

File: tests/installation/grub_test.pm [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+use strict;
+use base "basetest";
+use testapi;
+
+sub run() {
+    my $self = shift;
+    up_and_down();
+    assert_screen "grub2";
+    send_key 'esc';    # prevent grub2 timeout
```
Comment (okurz) [inline tests/installation/grub_test.pm]: repeating from the outdated version: Isn't the up_and_down already preventing the timeout?

File: tests/installation/grub_test.pm [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+use strict;
+use base "basetest";
+use testapi;
+
+sub run() {
+    my $self = shift;
+    up_and_down();
```
Comment (okurz) [inline tests/installation/grub_test.pm]: after the recent change by maxlin (press 'up' once) seems to be just enough. Is this really still necessary? We currently seem to have a situation where workers struggle with heavy load but have no problem to check the screen at least once during the grub timeout to abort the timeout after one successful screen match. I prefer to keep the current approach

File: tests/installation/grub_test.pm (hunk 4 above)
Comment (MalloZup) [inline tests/installation/grub_test.pm]: yes, but i tested that send_key 'esc' (without this) test are failing. 

I know you find this dirty, but if for test safety, i prefer dirty, but stable.

**Comment by okurz** [comment]
… and thy sins be forgiven thee. and you sinned again, brother, not using imperative mode all the way and not starting the subject line with capital letter. But not important, the inline comments are more relevant. let others give their two cents.

File: tests/installation/grub_test.pm (hunk 4 above)
Comment (okurz) [inline tests/installation/grub_test.pm]: So what does the "up_and_down" do then if you need to press escape after grub2 shows up?

File: tests/installation/grub_test.pm (hunk 5 above)
Comment (MalloZup) [inline tests/installation/grub_test.pm]: i have tested the maxlin approach, test continue to fails only with this, since you trust the needles matching. i would be more generous with 10 seconds more, than rather then 10 fails

**Comment by MalloZup** [comment]
ok,  i will change inline comments.

File: tests/installation/grub_test.pm (hunk 4 above)
Comment (MalloZup) [inline tests/installation/grub_test.pm]: up_and_down doesn't trust needle matching, so the escape at the end, is needed, if booting process is extremely looong, so we have both advantage, one from me , one from maxlin

Comment (coolo) [inline tests/installation/grub_test.pm]: If your machine seriously has trouble getting the needle within 8 seconds, get a better one.

Comment (MalloZup) [inline tests/installation/grub_test.pm]: it's look like that isn't my machine that has problem.
Here:
https://openqa.suse.de/tests/197318/modules/livecdreboot/steps/2/src
and :  https://openqa.suse.de/tests/197318/modules/livecdreboot/steps/1/src
. and there are so many example, with the stuff  with send key up doesn't work.

If you want that i change the code, i don't mind, but we still have other issues, because situation happens that we have slowly performance. ( i tried with my machine running 3 workers).

So  i know you don't like the dirty aspect of my solution, but i don't like the tests failures.
So before something will stable, ( in the api or backend, that's your focus) i will propose to take this workaround, to avoid such situations, that aren't my machine issues.

Comment (coolo) [inline tests/installation/grub_test.pm]: 197318 is a very bad example because we did not miss grub there - we found the rebootnow needle and pressed a key to stop the countdown and changed to tty2 - but ended up in the grub screen. That has very little to do with what I'm objecting with

**Comment by MalloZup** [comment]
ok, i removed the up_and_down function. we will see. 
So i will prepare the PR for branch SLE12/11
